src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -874,11 +874,15 @@
          :headers {"Content-Type" "application/octet-stream"}
          :body (get-code-content environment code-id file-path)}))))
 
+(defn valid-env-name?
+  [string]
+  (re-matches #"\w+" string))
```

Comment:
I don't know if this matters, but the ruby version of this uses `\A` and `\Z` to prevent against multiline inputs: https://github.com/puppetlabs/puppet/blob/c6412ea369aa1613ca166b10fb819fcf6130814b/lib/puppet/node/environment.rb#L127 Does this version do the same?

Reply:
My understanding is that Java regexes are anchored from start of string to end of string, newlines inclusive, by default, lemme double check in my repl.

annotated test in repl:
```
dev-tools=> (def wlines "foo\nfoo")
#'dev-tools/wlines

;; prn, and most of clojure string processing doesn't do anything special with these characters
dev-tools=> (prn wlines)
"foo\nfoo"
nil

;; but split-lines recognizes them
dev-tools=> (clojure.string/split-lines wlines)
["foo" "foo"]

;; regexes are processed across the whole string irrespective of special characters
dev-tools=> (re-matches #"\w+" "fo\nfo")
nil

dev-tools=> (re-matches #"\w+" "fofo")
"fofo"
```